packages/virtual-core/src/index.ts

```diff
@@ -371,14 +386,26 @@ export class Virtualizer<TScrollElement = unknown, TItemElement = unknown> {
       for (let i = min; i < count; i++) {
         const key = getItemKey(i)
         const measuredSize = measurementsCache[key]
-        const start = measurements[i - 1]
-          ? measurements[i - 1]!.end
-          : paddingStart
+
         const size =
           typeof measuredSize === 'number'
             ? measuredSize
             : this.options.estimateSize(i)
-        const end = start + size
+
+        let start
+        let end
+        if (reverse) {
+          end = measurements[i - 1]
+            ? measurements[i - 1]!.start
```

Comment:
Generally feel a bit hesitant about using `!`.. is there another way we can write this? Maybe wrap with an `if (measurements[i -1])` and then add some kind of error case?